src/reanimated2/platform-specific/webAnimations.ts

```diff
@@ -0,0 +1,278 @@
+import { Animations, AnimationsData, WebEasings } from './webAnimationsData';
+import type { AnimationTypes } from './webAnimationsData';
+import { parseAnimationObjectToKeyframe } from './animationParser';
+import type { TransformProperties } from './animationParser';
+import type { TransformsStyle } from 'react-native';
+import type {
+  BaseAnimationBuilder,
+  EntryExitAnimationFunction,
+  ILayoutAnimationBuilder,
+} from '../layoutReanimation';
+
+export const WEB_ANIMATIONS_ID = 'webAnimationsStyle';
+
+export interface AnimationConfig {
+  animationName: string;
+  duration: number;
+  delay: number;
+  easing: string;
+  callback: (() => void) | null;
+}
+
+interface CustomConfig {
+  easingV?: () => number;
+  durationV?: number;
+  delayV?: number;
+  randomizeDelay?: boolean;
+  callbackV?: () => void;
+}
+
+type ConfigType =
+  | BaseAnimationBuilder
+  | ILayoutAnimationBuilder
+  | typeof BaseAnimationBuilder
+  | EntryExitAnimationFunction
+  | Keyframe
+  | CustomConfig;
+
+/**
+ *  Creates `HTMLStyleElement`, inserts it into DOM and then inserts CSS rules into the stylesheet.
+ *  If style element already exists, nothing happens.
+ */
+export function insertWebAnimations(): void {
+  if (document.getElementById(WEB_ANIMATIONS_ID) !== null) {
+    return;
+  }
+
+  const style = document.createElement('style');
+  style.id = WEB_ANIMATIONS_ID;
+
+  document.head.appendChild(style);
+
+  for (const animationType in Animations) {
+    style.sheet?.insertRule(Animations[animationType as AnimationTypes].style);
+  }
+}
+
+/**
+ *  Modifies default animation by preserving transformations that given element already contains.
+ *
+ * @param animationName Name of the animation to be modified (e.g. `FadeIn`).
+ * @param existingTransform Transform values that element already contains.
+ * @returns Animation parsed to keyframe string.
+ */
+export function createAnimationWithExistingTransform(
+  animationName: string,
+  existingTransform: NonNullable<TransformsStyle['transform']>
+): string {
+  if (!(animationName in Animations)) {
+    return '';
+  }
+
+  const keyframeName = generateRandomKeyframeName();
+  const newAnimationData = structuredClone(AnimationsData[animationName]);
+
+  newAnimationData.name = keyframeName;
+
+  if (typeof existingTransform === 'string') {
+    throw new Error(
+      '[Reanimated] String transform is not currently supported.'
```

Comment:
```suggestion
      '[Reanimated] String transform is currently unsupported.'
```

Reply:
Done in [7f8735d](https://github.com/software-mansion/react-native-reanimated/pull/4805/commits/7f8735deb90c38024a96cbe8a49260758ee8097f)